File: HistorialAPI/src/main/java/com/clinica/api/historial_service/repository/HistorialRepository.java
package com.clinica.api.historial_service.repository;

import com.clinica.api.historial_service.model.Historial;
import java.util.List;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface HistorialRepository extends JpaRepository<Historial, Long> {

    List<Historial> findByIdUsuario(Long idUsuario);

    List<Historial> findByIdDoctor(Long idDoctor);
}

File: historial-service/bin/main/com/clinica/api/historial_service/repository/HistorialRepository.kt
package com.clinica.api.historial_service.repository

import com.clinica.api.historial_service.model.Historial
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface HistorialRepository : JpaRepository<Historial, Long> {
    fun findByUsuarioId(usuarioId: Long): List<Historial>
}
